src/block-components/block-link/index.js

```diff
@@ -23,7 +23,7 @@ BlockLink.Content = props => {
 	return (
 		<Link.Content
 			className="stk-block-link stk--transparent-overlay"
-			href={ attributes.blockLinkUrl }
+			href={ href || attributes.blockLinkUrl }
```

Comment:
May be better to rename the prop from `href` to `defaultHref` because of how the prop works.